kalite/distributed/static/js/distributed/rating/views.js

```diff
@@ -0,0 +1,263 @@
+            self.star_view_2.remove();
+            self.render3rd();
+        })
+    },
+
+    render3rd: function() {
+        this.star_view_3 = this.add_subview(StarView, {
+            el: this.$("#star-container-3"),
+            model: this.model,
+            rating_attr: "rating3"
+        });
+        var self = this;
+        this.model.once("change:rating3", function() {
+            self.star_view_3.remove();
+            self.renderTextArea();
+        })
+    },
+
+    renderTextArea: function() {
+        this.$(".rating-submit").show();
+        this.$(".rating-skip").show();
+        this.text_view = this.add_subview(TextView, {
+            el: this.$("#text-container"),
+            model: this.model
+        });
+        var self = this;
+        // Wrap in _.once, since it could potentially be called twice by different callbacks
+        var callback = _.once(function() {
+            self.text_view.remove();
+            self.renderAll();
+        });
+        this.$(".rating-submit").one("click", function() {
+            // Update the model explicitly before removing the view, since the callback for changed text is debounced
+            self.text_view.model.set(self.text_view.text_attr, self.text_view.$(".rating-text-feedback")[0].value);
+            callback();
+        });
+        this.$(".rating-skip").one("click", function() {
+            self.text_view.model.set(self.text_view.text_attr, "");
+            callback();
+        });
+    },
+
+    renderAll: function() {
+        /*
+            Renders itself, then attaches all subviews.
+            Called when: 1) The view's model is fetched successfully or
+                         2) The view's model is not fetched, and the user finishes filling out the new form.
+        */
+        this.$el.html(this.template());
+        // Explicitly hide/display desired buttons, since we re-render the template
+        this.$(".rating-skip").hide();
+        this.$(".rating-edit").hide();
+        this.star_view_1 = this.add_subview(StarView, {
+            el: this.$("#star-container-1"),
+            model: this.model,
+            rating_attr: "rating1"
+        });
+        this.star_view_2 = this.add_subview(StarView, {
+            el: this.$("#star-container-2"),
+            model: this.model,
+            rating_attr: "rating2"
+        });
+        this.star_view_3 = this.add_subview(StarView, {
+            el: this.$("#star-container-3"),
+            model: this.model,
+            rating_attr: "rating3"
+        });
+        this.text_view = this.add_subview(TextView, {
+            el: this.$("#text-container"),
+            model: this.model,
+            rating_attr: "text"
+        });
+        this.submit_rating();
+    },
+
+    submit_rating: function() {
+        this.text_view.toggle_disabled(true);
+        this.$(".rating-submit").hide();
+        this.$(".rating-edit").show();
+        this.model.save();

```

Comment:
The only thing I can think of is to give the user some feedback e.g. a message with `Your rating was not submitted! :(`